File: python/7_kyu/make_acronym.py
# makeAcronym
# https://www.codewars.com/kata/557efeb04effce569d000022/train/python

def make_acronym(phrase):
    if phrase == '':
        return phrase
    elif isinstance(phrase, str):
        phrase = phrase.split(" ")
        
        for i in phrase:
            if not i.isalpha():
                return "Not letters"

        ans = []

        for i in phrase:
            ans.append(i[0].upper())
        
        return ''.join(ans)
    else:
        return "Not a string"

File: javascript/8_kyu/makeAcronym.js
// makeAcronym
// https://www.codewars.com/kata/557efeb04effce569d000022/train/javascript

var makeAcronym = function(string){
    if (!(typeof string === 'string')) {
      return 'Not a string';
    }

    var ans = [];
    var lis = string.split(' ');

    for (var i = 0; i < lis.length; i++) {
        if (/^[a-zA-Z]+$/.test(lis[i]) == true) {
            ans.push(lis[i][0].toUpperCase());
        } else {
            return 'Not letters';
        }
    }
    return ans.join('');
};

console.log(makeAcronym('Hello codewarrior'));